tests/groupby_test.py

```diff
@@ -118,7 +118,7 @@ def run_test(levels, verbose=False):
 class GroupByTest(ArkoudaTest): 
 
     # https://github.com/mhmerrill/arkouda/issues/365
-    @unittest.skip
+    #@unittest.skip
```

Comment:
Still fails for me ~1/20 trials. There's probably a one-liner for this, but I'm just testing with:

```bash
#!/usr/bin/env bash

for run in {1..100}; do
  python3 -m pytest tests/groupby_test.py
  if [ $? -ne 0 ]; then
    exit 1
  fi
done
```

Reply:
@ronawho can you look for a message on the failure which looks like:
`Different values (abs diff = nan)`
this is what is printed when the answers don't match between pandas and arkouda.
which reduction is failing? Is it always the same one?
for me it was:
```
Doing aggregate(int64, prod)
Different values (abs diff = nan)
```
I ran your script and failed after 15 successes

Reply:
It's not clear to me where the actual failure occurs, but I do see:

```
Doing aggregate(int64, prod)
Different values (abs diff = nan)
```

I updated my script to just print failures to a file:

```bash
for run in {1..100}; do
  python3 -m pytest tests/groupby_test.py &> out.txt
  if [ $? -ne 0 ]; then
    cat out.txt >> failures.txt
    echo "Failure"
  fi
done
```

And here's the log for 4 failures: [failures.txt](https://github.com/mhmerrill/arkouda/files/4940891/failures.txt)

Reply:
@reuster986 @mcdobe100 it seems we did not implement `skipna` for the segmented product reduction. Or for segmented `argmin` or `argmax`

https://github.com/mhmerrill/arkouda/blob/f3e75a28ea47e29cb91c7ecc53e85da0161443e8/src/ReductionMsg.chpl#L315-L325
https://github.com/mhmerrill/arkouda/blob/f3e75a28ea47e29cb91c7ecc53e85da0161443e8/src/ReductionMsg.chpl#L588-L607

Reply:
@reuster986 it occurred to me that the log(0) is undefined, what if zero is sneaking into the computation. That would produce `NaN` wouldn't it. Maybe the algorithm here takes care of this but I keep seeing NaN from the check routine in the test.

Reply:
@reuster986 @ronawho @mcdobe100 so on further investigation it is the `int64` code path where the failures occur, if I add `+1` to the integer data the test passed 100 time in a row.

Reply:
ok now it passed 200 times in a row.

Reply:
Ah, so we probably just need logic that checks for zeros in the segment and short-circuits them. I will work on this.

Reply:
@mhmerrill regarding your latest commit, I think we should keep the zeros in the int64 test data. The `prod` reduction needs to be able to handle zeros -- I'm writing logic for this now, and I would like the test data to cover that case.

Reply:
@reuster986 I was going to put an issue in to cover this and decouple the logic fix from this PR so we would not delay these changes. Unless you are going to get the logic change done and tested... today or tomorrow.

Reply:
@mhmerrill Yes, that is correct, we only implemented `skipna` for `min`, `max`, `sum`, and `mean`. Would you like it to be extended to those operations as well? The reason we only went with those 4 is that the original intent for `skipna` was motivated by Roland's use, which only required the use of those 4, but I think it makes sense to have it implemented for all of them if you want that done as well.

Reply:
@mcdobe100 The code I'm testing right now includes `skipna` for `prod`. Since I was writing code for handling zeros, it was relatively easy to do the same for NaNs at the same time.

Reply:
I think I discovered a bug in pandas. When doing `groupby` followed by `.aggregate('prod')`, pandas will sometimes compute a `nan` when the product should be zero. I discovered this by manually generating a dataframe with int64 columns, doing a groupby-aggregate with `prod`, and finding a group with `nan` product. Then, I manually sliced out the group from the original dataframe and computed the true product, which was 0. 

I believe the pandas bug is the root cause of the intermittent test failures. In the latest commit, I modified `groupby_test.py` to replace `nan` with 0 in the pandas `prod` result, and I ran the 100-fold test with no failures.